### PR TITLE
Irhold

### PR DESCRIPTION
**Pull request by ChrisVeigl**
I modified the IR-play function so that it does not use a polling loop, 
furthermore added new commands "AT IH" (hold IR code replay) and "AT IS" (stop IR code replay) for repetition of the IR-code as long as a user action persists.

In the GUI, a combobox-based selection for existing IR-codes has beed added so that IR-code names do not have to be typed manually anymore

The IR-play and IR-hold functions should be tested with other devices, a first test with a TV-set was successful.

### CODE REVIEW

File: FLipWare/infrared.cpp [hunk 1]
```diff
@@ -273,4 +285,65 @@ void set_IR_timeout(uint16_t tout_ms)
    edge_timeout = (uint32_t)tout_ms * 1000;
 }
 
+
+// obsolete: old play code which used a loop
```
Comment (klues) [inline FLipWare/infrared.cpp]: old code could be deleted - git saves all the history, so old code in comments should not be needed.

File: FLipWare/infrared.cpp [hunk 2]
```diff
@@ -174,18 +182,51 @@ void delete_IR_command(char * name)
...
+    if (repeatCounter > 0) repeatCounter--;  
+    if (repeatCounter==0)  playTimer.end();  // stop time if last repetition done 
+                                             // note: repeatCounter is -1 for hold mode
+  }
+  else {
+    if (act_edge == edges)
+       playTimer.update(10000);  // 10 ms gap between code repetitions 
+    else playTimer.update(timings[act_edge]);  // get interval for current on/off phase 
+
+    if(output_state == HIGH) {
```
Comment (klues) [inline FLipWare/infrared.cpp]: maybe change `output_state` to `int` and simplify lines 205-214 to:
```
analogWrite(IR_LED_PIN, output_state);
if(output_state == 0) output_state = 128;
else output_state = 0;
```

**Comment by klues** [comment]
added two minor comments - otherwise this looks good for me and in my opinion could be merged.

**Comment by ChrisVeigl** [comment]
okay nice!  will integrate you suggestion !
did you actually try the function with diffferent IR-devices or compare the IR-output with the desired codes on the oscilloscope ? (I am not sure if the timings are accurate)

**Comment by klues** [comment]
no, I just reviewed the code, I didn't do tests with the oscilloscope. Maybe Fanny and Isis can do this? I don't know their github names, otherwise I could mention them here.

**Pull request merged**